### DISCUSSION TITLE
Overriding transforms on draggables when dealing with scrollbar/overflow/max-height situations.

### DISCUSSION THREAD

**Discussion by vincentjflorio** [clauderic/dnd-kit#1000] [General]
I am attempting to drag a draggable into one of a series of small droppables in a scrollable container, and if I leave the entire container open, at full height, all is well.

If instead I impose a max-height (about half its natural height) and an overflow-y hidden (to force a scroll, keeping half of the available droppables hidden), the draggable's transform goes haywire --- to the exact offset degree that the container's scrollTop has brought everything down.

I actually managed to use context to override the transform by that same amount, but there's a bit of flicker for the split-second where the transition from open space into the droppable begins. You can drag outside fine, you can drag inside pretty much right where we'd like, but for 100-200ms it is the same 'broken' offset.

From other bug reports / discussions here I believe the idea is that the transform is behaving 'naturally' but is there a way to smooth over the jump?

**Comment by vaynevayne** [comment]
same issue and had report an issue

**Comment by vincentjflorio** [comment]
@vaynevayne I did wind up sorting this out, mostly. I set up an onScroll event handler on the shortened, scrollable container (the one containing my droppables, stacked upon one another), and on a scroll I use e.target.scrollTop to determine how far down the user has scrolled and save it to state.

That state, as a prop, we'll call it 'droppableScrollTop', gets passed down to the draggable and I use it across a few steps to adjust the styles:

```
  const scrollDifference = transform?.y - droppableScrollTop
  const draggableTransformY = (droppableScrollTop && scrollDifference) ? scrollDifference : transform?.y
  const customTransformY = (over && paletteTransformY) ? draggableTransformY : transform?.y
  
   const style = transform ? {
    transform: `translate3d(${transform?.x}px, ${customTransformY > 0 && customTransformY < ### ? customTransformY : transform?.y}px, 0px)`,
    zIndex: 'auto',
  } : undefined;
```
where ### is however you choose to communicate how tall the scrollable droppable container is. I have other conditions (different things that use this component that aren't applicable) and in removing them, so as to not distract, it may mean the code above as-is is redundant, but that should give you a strong idea.
